src/attr/_make.py

```diff
@@ -1349,6 +1352,9 @@ def fmt_setter_with_converter(attr_name, value_var):
             else:
                 lines.append(fmt_setter(attr_name, arg_name))
 
+        if a.init is True and a.converter is None and a.type is not None:
```

Comment:
If there's a converter I don't include an annotation.  I don't think it's wise to try to follow the annotation chain.  Mypy itself has a hard time doing it too.

Reply:
Well you don't want the return value. You want the type of the first argument.  

I played around with it and it's pretty complicated because some converters don't have `__annotations` (e.g. `int`) and some have an implicit first argument like `self` or `cls`.  `inspect.signature` does the right thing but that call is pretty expensive.

Reply:
Fair enough - if performance is a consideration here I agree that skipping annotations is the best option when a converter is present.

Reply:
I agree with euresti, converters are hard. We can add those later, on the basis whether the **converter** is a type. If it is, just put the converter directly as an annotation. If it's not a type, then try to get the annotation of the first argument to the callable. But it's going to be a fun pull request on its own.

Reply:
But FWIW I'm not worried about performance here, it's essentially just a few additional dict lookups. To notice any adverse effect, a codebase would have to have thousands of fields with converters. And even then, we're only talking about slight increase in startup time.

Reply:
FYI `converter=<some type>` is actually pretty hard because what you actually want are all the types that the type accepts as a first argument.   (e.g. `int` would be `Union[float, str, int, bool]` and probably a bunch of other things.)

Reply:
Well, now we're opening a can of worms, which is asymmetric types, e.g. the setter accepts a broader spectrum of types than the getter. In __annotations__, there is no way to fully express this. The fields will always have the final type (the one after conversion). You can make unions for __init__'s arguments but that won't cover the case where users modify fields by assignment.

And, as you said, plucking the unions from types is going to require a registry for C types, including builtins. And if user classes aren't annotated, we know nothing besides the final type.

My intuition is to just handle final types all around and see what users say later.